kb/development/developer-builds.md

```diff
@@ -0,0 +1,727 @@
+```
+
+With mbedtls-prepare-build, specify paths relative to the build directory:
+
+```
+mbedtls-prepare-build -d build-suiteb --crypto-config-file=../configs/crypto-config-suite-b.h --config-file=../configs/config-suite-b.h
+```
+
+#### Use a configuration file extension
+
+You can select additional configuration files (effectively appended to `mbedtls/mbedtls_config.h` and `psa/crypto_config.h`) by setting the C preprocessor macros `MBEDTLS_USER_CONFIG_FILE` and `TF_PSA_CRYPTO_USER_CONFIG_FILE`.
+
+With CMake, specify paths relative to the current directory:
+
+```
+cmake -B build-suiteb -D MBEDTLS_USER_CONFIG_FILE=tests/configs/user-config-for-test.h
+```
+
+User config files are not supported by mbedtls-prepare-build. You can use `--config-set` and `--config-unset` to tweak individual options, or [set the variables manually](#use-a-different-configuration-file-manual-way).
+
+#### Use a different configuration file (manual way)
+
+You can select different configuration files by setting the C preprocessor macros `MBEDTLS_CONFIG_FILE` and `TF_PSA_CRYPTO_CONFIG_FILE` via `CFLAGS`. In both cases, the value is what is after `#include`, so it requires double quotes or angle brackets around the file name; beware of quoting. Furthermore the file name must be either absolute, or relative to a directory that is in the include path.
+
+With CMake (but note that modern versions of Mbed TLS have better support through CMake variables for the [main config files](#Use-a-different-configuration-file-simple-way) and [user config files](#use-a-configuration-file-extension)):
+
+```
+CFLAGS='-DMBEDTLS_CONFIG_FILE=\"../configs/config-tfm.h\" -Os' cmake -B build-tfm
+```
+
+With mbedtls-prepare-build:
+
+```
+mbedtls-prepare-build -d build-tfm --cflags '-DMBEDTLS_CONFIG_FILE=\"../configs/config-tfm.h\" -Os'
+```
+
+### Set individual options
+
+You can set additional boolean options, or change the values of non-boolean options, by passing `-D` to the compiler in `CFLAGS`. See “[Choose compiler and compiler flags manually](#choose-compiler-and-compiler-flags-manually)”. This method does not allow unsetting options that are enabled by default.
+
+With mbedtls-prepare-build, you can also use `--config-set`. [Completion](#command-line-completion) is available.
+
+```
+mbedtls-prepare-build -d build-custom --config-set=MBEDTLS_ECP_RESTARTABLE,MBEDTLS_ECP_WINDOW_SIZE=2
+```
+
+or
+
+```
+mbedtls-prepare-build -d build-custom --config-set=MBEDTLS_ECP_RESTARTABLE --config-set=MBEDTLS_ECP_WINDOW_SIZE=2
+```
+
+Limitation: at the time of writing, in Mbed TLS, mbedtls-prepare-build only tweaks options in `mbedtls_config.h`, so this doesn't work for crypto options with Mbed TLS 4.
+
+### Set and unset individual options
+
+CMake doesn't have a direct way to tweak individual config options. You can write `#define` and `#undef` directives in a header file and pass that as the [user config file](#use-a-configuration-file-extension). To run `config.py`, ???
```

Comment:
A build directory with a tweaked configuration is something I do very often. I don't know how to do it with CMake.